bundler/lib/bundler/lockfile_generator.rb

```diff
@@ -65,6 +66,22 @@ def add_dependencies
       end
     end
 
+    def add_checksums
+      out << "\nCHECKSUMS\n"
+      definition.resolve.sort_by(&:full_name).each do |spec|
+        checksum = spec.to_checksum if spec.respond_to?(:to_checksum)
```

Comment:
when would the spec not respond to `to_checksum` ?

Reply:
`LazySpecification` doesn't respond to `to_checksum`

Reply:
So it turns out it actually does, which is causing spec failures when git gems are being excluded